app/main/views.py

```diff
@@ -613,7 +614,14 @@ def user_list_page():
         search_result_email = User.query.filter(User.email.ilike('%' + entry + '%')).all()
         search_result_fname = User.query.filter(User.first_name.ilike('%' + entry.title()+ '%')).all()
         search_result_lname = User.query.filter(User.last_name.ilike('%' + entry.title() + '%')).all()
-        list_of_users = list(set(list_of_users_all) & set(search_result_email + search_result_fname + search_result_lname))
+        search_result_division = User.query.filter(User.division.ilike('%' + entry.title() + '%')).all()
```

Comment:
Should this work with partial matches? 

e.g. `PEN` or `pen` should match any `PENCIL` interns?